[2022-02-01 13:15:58,265] {taskinstance.py:1032} INFO - Dependencies all met for <TaskInstance: spark-test.spark_job manual__2022-02-01T13:15:57.116281+00:00 [queued]>
[2022-02-01 13:15:58,275] {taskinstance.py:1032} INFO - Dependencies all met for <TaskInstance: spark-test.spark_job manual__2022-02-01T13:15:57.116281+00:00 [queued]>
[2022-02-01 13:15:58,276] {taskinstance.py:1238} INFO - 
--------------------------------------------------------------------------------
[2022-02-01 13:15:58,276] {taskinstance.py:1239} INFO - Starting attempt 1 of 2
[2022-02-01 13:15:58,276] {taskinstance.py:1240} INFO - 
--------------------------------------------------------------------------------
[2022-02-01 13:15:58,285] {taskinstance.py:1259} INFO - Executing <Task(SparkSubmitOperator): spark_job> on 2022-02-01 13:15:57.116281+00:00
[2022-02-01 13:15:58,290] {standard_task_runner.py:52} INFO - Started process 87 to run task
[2022-02-01 13:15:58,295] {standard_task_runner.py:76} INFO - Running: ['***', 'tasks', 'run', 'spark-test', 'spark_job', 'manual__2022-02-01T13:15:57.116281+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/spark-dag.py', '--cfg-path', '/tmp/tmpxo8gk4zx', '--error-file', '/tmp/tmph3m8lw3k']
[2022-02-01 13:15:58,296] {standard_task_runner.py:77} INFO - Job 5: Subtask spark_job
[2022-02-01 13:15:58,366] {logging_mixin.py:109} INFO - Running <TaskInstance: spark-test.spark_job manual__2022-02-01T13:15:57.116281+00:00 [running]> on host 5adaf769da9d
[2022-02-01 13:15:58,436] {taskinstance.py:1426} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=***@***.com
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=spark-test
AIRFLOW_CTX_TASK_ID=spark_job
AIRFLOW_CTX_EXECUTION_DATE=2022-02-01T13:15:57.116281+00:00
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-02-01T13:15:57.116281+00:00
[2022-02-01 13:15:58,449] {base.py:79} INFO - Using connection to: id: spark_default. Host: spark://spark, Port: 7077, Schema: , Login: , Password: None, extra: {}
[2022-02-01 13:15:58,452] {spark_submit.py:360} INFO - Spark-Submit cmd: spark-submit --master spark://spark:7077 --conf spark.master=spark://spark:7077 --name Spark Hello World --verbose C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\spark-book-count.py C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\book.txt
[2022-02-01 13:15:58,532] {spark_submit.py:514} INFO - JAVA_HOME is not set
[2022-02-01 13:15:58,543] {taskinstance.py:1700} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1329, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1455, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1511, in _execute_task
    result = execute_callable(context=context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 182, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 446, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --conf spark.master=spark://spark:7077 --name Spark Hello World --verbose C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\spark-book-count.py C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\book.txt. Error code is: 1.
[2022-02-01 13:15:58,548] {taskinstance.py:1277} INFO - Marking task as UP_FOR_RETRY. dag_id=spark-test, task_id=spark_job, execution_date=20220201T131557, start_date=20220201T131558, end_date=20220201T131558
[2022-02-01 13:15:58,556] {standard_task_runner.py:92} ERROR - Failed to execute job 5 for task spark_job
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/task/task_runner/standard_task_runner.py", line 85, in _start_by_fork
    args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/cli.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/commands/task_command.py", line 298, in task_run
    _run_task_by_selected_method(args, dag, ti)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/commands/task_command.py", line 107, in _run_task_by_selected_method
    _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/commands/task_command.py", line 184, in _run_raw_task
    error_file=args.error_file,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py", line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1329, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1455, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1511, in _execute_task
    result = execute_callable(context=context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 182, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 446, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --conf spark.master=spark://spark:7077 --name Spark Hello World --verbose C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\spark-book-count.py C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\book.txt. Error code is: 1.
[2022-02-01 13:15:58,587] {local_task_job.py:154} INFO - Task exited with return code 1
[2022-02-01 13:15:58,614] {local_task_job.py:264} INFO - 0 downstream tasks scheduled from follow-on schedule check
